config/initializers/01_redis.rb

```diff
@@ -4,6 +4,9 @@
 REDIS_CONFIG = Rails.application.config_for(:redis).freeze
 # set the current global instance of Redis based on environment specific config
 
-$redis = Redis.new(REDIS_CONFIG[:redis].to_hash)
-
-Redis.exists_returns_integer = true
+$redis = if Rails.env.test?
+           require 'mock_redis'
+           MockRedis.new(url: REDIS_CONFIG[:redis][:url])
+         else
+           Redis.new(REDIS_CONFIG[:redis].to_h)
+         end
```

Comment:
```suggestion
$redis =
  if Rails.env.test?
    require 'mock_redis'
    MockRedis.new(url: REDIS_CONFIG[:redis][:url])
  else
    Redis.new(REDIS_CONFIG[:redis].to_h)
  end
```

I usually prefer this syntax so that I don't have weird indentation below.